temporalio/worker/replayer.py

```diff
@@ -125,25 +142,91 @@ async def replay_workflow(
         workflow_worker = _WorkflowWorker(
             bridge_worker=lambda: bridge_worker,
             namespace=self._config["namespace"],
-            task_queue=started_event.workflow_execution_started_event_attributes.task_queue.name,
+            task_queue=tq,
             workflows=self._config["workflows"],
             workflow_task_executor=self._config["workflow_task_executor"],
             workflow_runner=self._config["workflow_runner"],
             data_converter=self._config["data_converter"],
             interceptors=self._config["interceptors"],
             debug_mode=self._config["debug_mode"],
-            fail_on_eviction=True,
+            on_eviction_hook=self._replayer_eviction_hook(
+                fail_fast=self._config["fail_fast"],
+                pusher=pusher,
+                current_results=current_run_results,
+            ),
         )
 
-        # Run it
-        try:
-            await workflow_worker.run()
-        finally:
-            # We must finalize shutdown here
+        async def history_feeder():
+            try:
+                async for history in histories:
+                    # Extract workflow started event
+                    started_event = next(
+                        (
+                            e
+                            for e in history.events
+                            if e.HasField("workflow_execution_started_event_attributes")
+                        ),
+                        None,
+                    )
+                    if not started_event:
+                        raise ValueError("Started event not found")
+
+                    as_history_proto = temporalio.api.history.v1.History(
+                        events=history.events
+                    )
+                    await pusher.push_history(
+                        history.workflow_id, as_history_proto.SerializeToString()
+                    )
+            finally:
+                pusher.close()
+
+        async def runner():
+            # Run it
             try:
-                await bridge_worker.finalize_shutdown()
-            except Exception:
-                logger.warning("Failed to finalize shutdown", exc_info=True)
+                await workflow_worker.run()
+            finally:
+                # We must finalize shutdown here
+                try:
+                    await bridge_worker.finalize_shutdown()
+                except Exception:
+                    logger.warning("Failed to finalize shutdown", exc_info=True)
+
+        await asyncio.gather(history_feeder(), runner())
```

Comment:
A note from https://docs.python.org/3/library/asyncio-task.html#asyncio.gather:

> If return_exceptions is False (default), the first raised exception is immediately propagated to the task that awaits on gather(). Other awaitables in the aws sequence won’t be cancelled and will continue to run.
>
> If return_exceptions is True, exceptions are treated the same as successful results, and aggregated in the result list.

So that means if one of those tasks fails, the other is not cancelled/stopped by default. If core is not causing the other to fail when one is done, may need to account for that.

Reply:
Yeah, that's fine - and the tests show me it's working OK with fail fast anyway.